data/mods/gen4/moves.ts

```diff
@@ -742,7 +742,11 @@ export const Moves: import('../../../sim/dex-moves').ModdedMoveDataTable = {
 			this.add('-activate', source, 'move: Heal Bell');
 			const allies = [...target.side.pokemon, ...target.side.allySide?.pokemon || []];
 			for (const ally of allies) {
-				if (!ally.hasAbility('soundproof')) ally.cureStatus(true);
+				if (ally.hasAbility('soundproof')) {
+					if (ally.isActive) this.add('-immune', ally, '[from] ability: Soundproof');
```

Comment:
```suggestion
					if (ally.isActive) this.add('-immune', ally, '[from] ability: Soundproof');
```
does gen 4 have the immunity message in doubles?

Reply:
![image](https://github.com/user-attachments/assets/9f3096e6-7848-4438-876d-69d01f57f312)